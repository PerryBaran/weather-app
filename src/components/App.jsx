import React from "react";
import "../styles/App.css";
import PropTypes from "prop-types";
import LocationDetails from "./LocationDetails";
import ForecastSummaries from "./ForecastSummaries";

function App(props) {
  const {
    data: { forecasts, location },
  } = props;

  return (
    <div className="weather-app">
      <LocationDetails location={location} />
      <ForecastSummaries forecasts={forecasts} />
    </div>
  );
}

App.propTypes = {
  data: PropTypes.shape({
    forecasts: PropTypes.arrayOf(
      PropTypes.shape({
        date: PropTypes.number,
        description: PropTypes.string,
        icon: PropTypes.string,
        temperature: PropTypes.shape({
          max: PropTypes.number,
          min: PropTypes.number,
        }),
      })
    ),
    location: PropTypes.shape({
      city: PropTypes.string,
      country: PropTypes.string,
    }),
  }).isRequired,
};

export default App;
