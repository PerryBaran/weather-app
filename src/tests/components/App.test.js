import React from "react";
import { render, screen } from "@testing-library/react";
import axios from "axios";
import mockData from "../data/forecastData.json";
import App from "../../components/App";

describe("App", () => {
  test("snapshot", () => {
    jest.spyOn(axios, "get").mockResolvedValue(mockData);
    const { asFragment } = render(<App />);

    expect(asFragment()).toMatchSnapshot();
  });

  test("Search results rendered on app load ", async () => {
    jest.spyOn(axios, "get").mockResolvedValue(mockData);
    render(<App />);

    expect(await screen.findByText("Leeds, GB")).toBeInTheDocument();
  });
});
