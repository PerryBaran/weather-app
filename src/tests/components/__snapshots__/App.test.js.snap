// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot 1`] = `
<DocumentFragment>
  <div
    class="weather-app"
  >
    <div
      class="top-bar"
    >
      <div
        class="search-form"
      >
        <input
          type="text"
        />
        <button
          type="submit"
        >
          Search
        </button>
      </div>
      <button
        class="options-menu-button"
        type="button"
      >
        <img
          alt="options menu"
          src="menu.png"
        />
      </button>
      <div
        class="options-menu-form__container"
      >
        <form
          class="options-menu-form hidden"
          name="options menu form"
        >
          <h3>
            Options
          </h3>
          <div
            class="options-menu-form__units"
          >
            <h4>
              Units:
            </h4>
            <label
              class="options-menu-form__label"
              for="standardRadio"
            >
              Scientific
              <input
                checked=""
                class="options-menu-form__radio"
                id="standardRadio"
                name="units"
                type="radio"
                value="standard"
              />
            </label>
            <label
              class="options-menu-form__label"
              for="metricRadio"
            >
              Metric
              <input
                class="options-menu-form__radio"
                id="metricRadio"
                name="units"
                type="radio"
                value="metric"
              />
            </label>
            <label
              class="options-menu-form__label"
              for="imperialRadio"
            >
              Imperial
              <input
                class="options-menu-form__radio"
                id="imperialRadio"
                name="units"
                type="radio"
                value="imperial"
              />
            </label>
          </div>
          <button
            class="options-menu-form__apply"
            type="submit"
          >
            Apply
          </button>
        </form>
      </div>
    </div>
    <h1>
      Leeds, GB
    </h1>
    <div
      class="forecast-daily-summaries"
    >
      <div
        class="forecast-daily-summary"
        data-testid="forecast-daily-summary"
      >
        <p>
          Thu 
          <br />
           Nov 03
        </p>
        <div
          class="forecast-daily-summary__icon"
          data-testid="forecast-icon"
        >
          <i
            class="wi wi-day-cloudy"
            name="owm"
          />
           
        </div>
        <p
          class="forecast-daily-summary__description"
        >
          scattered 
          <br />
          clouds
        </p>
        <p>
          10.3K
        </p>
        <button
          class="forecast-daily-summary__more-details"
          type="button"
        >
          More Details
        </button>
      </div>
      <div
        class="forecast-daily-summary"
        data-testid="forecast-daily-summary"
      >
        <p>
          Fri 
          <br />
           Nov 04
        </p>
        <div
          class="forecast-daily-summary__icon"
          data-testid="forecast-icon"
        >
          <i
            class="wi wi-day-cloudy"
            name="owm"
          />
           
        </div>
        <p
          class="forecast-daily-summary__description"
        >
          scattered 
          <br />
          clouds
        </p>
        <p>
          8.3K
        </p>
        <button
          class="forecast-daily-summary__more-details"
          type="button"
        >
          More Details
        </button>
      </div>
      <div
        class="forecast-daily-summary"
        data-testid="forecast-daily-summary"
      >
        <p>
          Sat 
          <br />
           Nov 05
        </p>
        <div
          class="forecast-daily-summary__icon"
          data-testid="forecast-icon"
        >
          <i
            class="wi wi-day-cloudy"
            name="owm"
          />
           
        </div>
        <p
          class="forecast-daily-summary__description"
        >
          scattered 
          <br />
          clouds
        </p>
        <p>
          8.8K
        </p>
        <button
          class="forecast-daily-summary__more-details"
          type="button"
        >
          More Details
        </button>
      </div>
      <div
        class="forecast-daily-summary"
        data-testid="forecast-daily-summary"
      >
        <p>
          Sun 
          <br />
           Nov 06
        </p>
        <div
          class="forecast-daily-summary__icon"
          data-testid="forecast-icon"
        >
          <i
            class="wi wi-day-cloudy"
            name="owm"
          />
           
        </div>
        <p
          class="forecast-daily-summary__description"
        >
          scattered 
          <br />
          clouds
        </p>
        <p>
          8.7K
        </p>
        <button
          class="forecast-daily-summary__more-details"
          type="button"
        >
          More Details
        </button>
      </div>
      <div
        class="forecast-daily-summary"
        data-testid="forecast-daily-summary"
      >
        <p>
          Mon 
          <br />
           Nov 07
        </p>
        <div
          class="forecast-daily-summary__icon"
          data-testid="forecast-icon"
        >
          <i
            class="wi wi-day-cloudy"
            name="owm"
          />
           
        </div>
        <p
          class="forecast-daily-summary__description"
        >
          broken 
          <br />
          clouds
        </p>
        <p>
          12.2K
        </p>
        <button
          class="forecast-daily-summary__more-details"
          type="button"
        >
          More Details
        </button>
      </div>
      <div
        class="forecast-daily-summary"
        data-testid="forecast-daily-summary"
      >
        <p>
          Tue 
          <br />
           Nov 08
        </p>
        <div
          class="forecast-daily-summary__icon"
          data-testid="forecast-icon"
        >
          <i
            class="wi wi-day-rain"
            name="owm"
          />
           
        </div>
        <p
          class="forecast-daily-summary__description"
        >
          moderate 
          <br />
          rain
        </p>
        <p>
          11.1K
        </p>
        <button
          class="forecast-daily-summary__more-details"
          type="button"
        >
          More Details
        </button>
      </div>
    </div>
    <div
      class="forecast-details"
    >
      <h3>
        Thu Nov 03 2022
      </h3>
      <div
        class="forecast-breakdown"
      >
        <div
          class="forecast-breakdown__right"
        >
          <div
            data-testid="forecast-icon"
          >
            <i
              class="wi wi-day-cloudy"
              name="owm"
            />
          </div>
          <p>
            scattered clouds
          </p>
        </div>
        <div
          class="forecast-breakdown__left"
        >
          <p
            class="forecast-breakdown__temp"
          >
            10.3
            <sup>
              K
            </sup>
          </p>
          <div>
            <h4>
              Min: 
            </h4>
            <p>
              8.1K
            </p>
          </div>
          <div>
            <h4>
              Max: 
            </h4>
            <p>
              12.3K
            </p>
          </div>
          <div>
            <h4>
              Humidity: 
            </h4>
            <p>
              70.7%
            </p>
          </div>
          <div>
            <h4>
              Wind Speed: 
            </h4>
            <p>
              1.1m/s
            </p>
          </div>
        </div>
      </div>
      <div
        class="forecast-trihourly-summaries"
      >
        <div
          class="forecast-trihourly-summary"
          data-testid="forecast-trihourly-summary"
        >
          <p
            class="forecast-trihourly-summary__time"
          >
            00:00
          </p>
          <p
            class="forecast-trihourly-summary__no-data"
          >
            No 
            <br />
            Data
          </p>
        </div>
        <div
          class="forecast-trihourly-summary"
          data-testid="forecast-trihourly-summary"
        >
          <p
            class="forecast-trihourly-summary__time"
          >
            03:00
          </p>
          <p
            class="forecast-trihourly-summary__no-data"
          >
            No 
            <br />
            Data
          </p>
        </div>
        <div
          class="forecast-trihourly-summary"
          data-testid="forecast-trihourly-summary"
        >
          <p
            class="forecast-trihourly-summary__time"
          >
            06:00
          </p>
          <p
            class="forecast-trihourly-summary__no-data"
          >
            No 
            <br />
            Data
          </p>
        </div>
        <div
          class="forecast-trihourly-summary"
          data-testid="forecast-trihourly-summary"
        >
          <p
            class="forecast-trihourly-summary__time"
          >
            09:00
          </p>
          <p
            class="forecast-trihourly-summary__no-data"
          >
            No 
            <br />
            Data
          </p>
        </div>
        <div
          class="forecast-trihourly-summary"
          data-testid="forecast-trihourly-summary"
        >
          <p
            class="forecast-trihourly-summary__time"
          >
            12:00
          </p>
          <p
            class="forecast-trihourly-summary__no-data"
          >
            No 
            <br />
            Data
          </p>
        </div>
        <div
          class="forecast-trihourly-summary"
          data-testid="forecast-trihourly-summary"
        >
          <p
            class="forecast-trihourly-summary__time"
          >
            15:00
          </p>
          <div
            class="forecast-trihourly-summary__icon"
            data-testid="forecast-icon"
          >
            <i
              class="wi wi-day-cloudy"
              name="owm"
            />
          </div>
          <div
            class="forecast-trihourly-summary__description"
          >
            <p>
              scattered 
              <br />
              clouds
            </p>
          </div>
          <p
            class="forecast-trihourly-summary__temp"
          >
            12.3K
          </p>
        </div>
        <div
          class="forecast-trihourly-summary"
          data-testid="forecast-trihourly-summary"
        >
          <p
            class="forecast-trihourly-summary__time"
          >
            18:00
          </p>
          <div
            class="forecast-trihourly-summary__icon"
            data-testid="forecast-icon"
          >
            <i
              class="wi wi-day-rain"
              name="owm"
            />
          </div>
          <div
            class="forecast-trihourly-summary__description"
          >
            <p>
              light 
              <br />
              rain
            </p>
          </div>
          <p
            class="forecast-trihourly-summary__temp"
          >
            10.4K
          </p>
        </div>
        <div
          class="forecast-trihourly-summary"
          data-testid="forecast-trihourly-summary"
        >
          <p
            class="forecast-trihourly-summary__time"
          >
            21:00
          </p>
          <div
            class="forecast-trihourly-summary__icon"
            data-testid="forecast-icon"
          >
            <i
              class="wi wi-day-cloudy"
              name="owm"
            />
          </div>
          <div
            class="forecast-trihourly-summary__description"
          >
            <p>
              scattered 
              <br />
              clouds
            </p>
          </div>
          <p
            class="forecast-trihourly-summary__temp"
          >
            8.1K
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
